### ISSUE TITLE
Consider using ShortIntegerRef in caster

### ISSUE THREAD

**Issue by kendonB** [dictation-toolbox/Caster#535]
Ref: https://github.com/dictation-toolbox/dragonfly/pull/71

Picking up from #174:

To switch things over from caster's IntegerRefST, we must first understand what it does compared to dragonfly's IntegerRef.

https://github.com/dictation-toolbox/Caster/blob/c21a9f8034b432ad3247362d3c92c4dc0e7a21bd/castervoice/lib/dfplus/additions.py#L61

@synkarius or @Versatilus, do you know/remember what the IntegerRefST class provides over dragonfly's equivalent?

**Comment by LexiconCode** [comment]
I'm not sure if it's tied to SelectiveAction. https://github.com/dictation-toolbox/Caster/blob/c5074e2a8b7de348cbfa80a4686e7e2d5db2ed9a/caster/lib/ccr/javascript/javascript.py#L64

From the looks of it SelectiveAction might be the equivalent to dragonfly ContextAction.

**Comment by lexxish** [comment]
This will also be needed in order to decouple https://github.com/dictation-toolbox/Caster/issues/533 from Caster

**Comment by mrob95** [comment]
The only difference between `IntegerRefST` and `IntegerRef` is that ST allows the alternative number system to be used if the flag is activated in settings, from alphanumeric:
```
def numbers_list_1_to_9():
    result = ["one", "torque", "traio", "fairn", "faif", "six", "seven", "eigen", "nine"]
    if not settings.SETTINGS["miscellaneous"]["integer_remap_opt_in"]:
        result[1] = "two"
        result[2] = "three"
        result[3] = "four"
        result[4] = "five"
        result[7] = "eight"
    return result


def numbers_map_1_to_9():
    result = {}
    l = numbers_list_1_to_9()
    for i in range(0, len(l)):
        result[l[i]] = i + 1
    return result
```

Does anybody actually use these? I don't know. I think if `ShortIntegerRef` is preferred then it would be fine to replace `IntegerRefST`.

**Comment by LexiconCode** [comment]
Nope, I have no one speaks up let's depreciate it

**Comment by kendonB** [comment]
Asked on gitter and no one spoke up. I definitely think we should deprecate; however, I'd vote for removal since we have no evidence that it's being used and it would be more awkward to code up a ShortIntegerRefST in the meantime. This is also more something that belongs in Dragonfly. Thoughts?

**Comment by mrob95** [comment]
It would probably be possible to create a mechanism within dragonfly so that the integer classes can be more easily subclassed to modify pronunciation etc. will look into it if the need arises.

**Comment by kendonB** [comment]
Closed via #555.
